From: Chris Lamb <lamby@debian.org>
Date: Thu, 10 Dec 2015 17:03:23 +0000
Subject: reproducible-build

Remove timestamps from the build system.  Jacal can then be built
reproducibly using the reproducible build's reproducible toolchain.

See https://wiki.debian.org/ReproducibleBuilds

See bugs.debian.org/807587
---
 Makefile | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f0bb7ba..b6dd6e2 100644
--- a/Makefile
+++ b/Makefile
@@ -230,6 +230,12 @@ t:
 #### Stuff for maintaining JACAL below ####
 
 ver = $(VERSION)
+fmt = '%B %Y'
+ifdef SOURCE_DATE_EPOCH
+jacaldate = $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "+$(fmt)"  2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "+$(fmt)" 2>/dev/null || date -u "+$(fmt)")
+else
+jacaldate = $(shell date "+$(fmt)")
+endif
 
 lint:	$(cfiles) $(sfiles) $(gfiles)
 	sclint $(cfiles) $(sfiles) $(gfiles)
@@ -336,7 +342,7 @@ updates:
 
 new:	updates
 	echo @set JACALVERSION $(ver) > version.txi
-	echo @set JACALDATE `date +"%B %Y"` >> version.txi
+	echo @set JACALDATE $(jacaldate) >> version.txi
 	echo `date -I` \ Aubrey Jaffer \ \<`whoami`@`hostname`\>> change
 	echo>> change
 	echo \	\* toploads.scm \(*jacal-version*\): Bumped from $(VERSION) to $(ver).>>change
